### PR TITLE
Update Go to 1.18

### PR DESCRIPTION
**Pull request by swenson**
From 1.17.12

1.18.5 was just released, but not all packages have been updated, so I
went with 1.18.4

### CODE REVIEW

File: changelog/16522.txt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+```release-note:change
```
Comment (cipherboy) [inline changelog/16522.txt:1]: I think you need to move to the [correct changelog format](https://github.com/hashicorp/vault/blob/main/changelog/_go-ver-1120.txt).

Comment (swenson) [reply changelog/16522.txt:1]: hmm. I don't see any differences other than the name?

Comment (cipherboy) [reply changelog/16522.txt:1]: Yeah, the name is different, that's why the CI `Check Changelog` pass failed :-) 

(See the internal release process confluence for some discussion of the naming structure for Go version bumps in particular).

Comment (swenson) [reply changelog/16522.txt:1]: ah, I see: `Please use the format changelog/go-ver-<VAULT_VERSION_WITHOUT_DOTS>.txt instead`

Comment (cipherboy) [reply changelog/16522.txt:1]: Oh hm, it needs an underscore in the example. Could've sworn I saw a PR from @mladlow to fix that that @stevendpclark merged, but now I don't see it, hmmmmmm

Comment (swenson) [reply changelog/16522.txt]: Done, though it is a bit confusing since there is `_go-ver-1120.txt` and now `go-ver-1120.txt`

Comment (cipherboy) [reply changelog/16522.txt]: Yeah, its so we only have one Go version number in the final release. :-)

Comment (swenson) [reply changelog/16522.txt]: Makes sense

**Review by cipherboy — APPROVED** [review]

File: changelog/16522.txt (hunk 1 above)
Comment (stevendpclark) [reply changelog/16522.txt]: Oops I missed the fact that it was done on vault-enterprise during my review *sigh*. We will need to bring this to OSS.

https://github.com/hashicorp/vault-enterprise/pull/3050/files

**Review by sarahethompson — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -71,7 +71,7 @@ Developing Vault
 
 If you wish to work on Vault itself or any of its built-in systems, you'll
 first need [Go](https://www.golang.org) installed on your machine. Go version
-1.17.12+ is *required*.
+1.8.5+ is *required*.
```
[the remark this answers is not in the document]
Comment (swenson) [reply README.md:74]: Ugh, you would not believe how often I've done this and it is has messed me up :)

File: api/go.mod [hunk 3]
```diff
@@ -1,6 +1,6 @@
 module github.com/hashicorp/vault/api
 
-go 1.17
```
[the remark this answers is not in the document]
Comment (swenson) [reply api/go.mod]: Good point. This isn't critical, so I'll just roll this back.

**Review by stevendpclark — APPROVED** [review]
👍

**Comment by swenson** [comment]
Thanks everyone!

**Pull request merged**